src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -777,32 +785,50 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
       }));
   };
 
+  /**
+   * The number of milliseconds time should be adjusted by when computing the
+   * "time" for a block.
+   */
   #timeAdjustment: number = 0;
 
   /**
    * Returns the timestamp, adjusted by the timeAdjustment offset, in seconds.
+   * @param precedingTimestamp - the timestamp of the block to be used as the
+   * time source if `timestampIncrement` is not "clock".
    */
-  #currentTime = () => {
-    return Math.floor((Date.now() + this.#timeAdjustment) / 1000);
+  #adjustedTime = (precedingTimestamp: Quantity) => {
+    const timeAdjustment = this.#timeAdjustment;
+    const timestampIncrement = this.#options.miner.timestampIncrement;
+    if (timestampIncrement === "clock") {
+      return Math.floor((Date.now() + timeAdjustment) / 1000);
+    } else {
+      return (
+        precedingTimestamp.toNumber() +
+        Math.floor(timeAdjustment / 1000) +
+        timestampIncrement.toNumber()
+      );
+    }
   };
 
   /**
-   * @param seconds -
+   * @param milliseconds - the number of milliseconds to adjust the time by.
+   * Negative numbers are treated as 0.
    * @returns the total time offset *in milliseconds*
    */
-  public increaseTime(seconds: number) {
-    if (seconds < 0) {
-      seconds = 0;
+  public increaseTime(milliseconds: number) {
+    if (milliseconds < 0) {
+      milliseconds = 0;
     }
-    return (this.#timeAdjustment += seconds);
+    return (this.#timeAdjustment += milliseconds);
   }
 
   /**
-   * @param seconds -
+   * @param newTime - the number of milliseconds to adjust the time by. Can be negative.
+   * @param sourceTime - the current time in milliseconds
```

Comment:
Isn't this the current time _or_ the timestamp of the latest block we're using as a reference?

Reply:
removed per your other comment.